orc8r/cloud/go/services/orchestrator/servicers/grpc_exporter_servicer.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Magma Authors.
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"magma/orc8r/cloud/go/services/metricsd/protos"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_model/go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+
+	edgeHubPB "github.com/facebookincubator/prometheus-edge-hub/grpc"
+)
+
+const (
+	grpcMaxTimeoutSec = 60
+	grpcMaxDelaySec   = 20
+	grpcMaxMsgSize    = 1024 * 1024 * 1024 // 1 Gb
+)
+
+type GRPCPushExporterServicer struct {
+	PushAddresses []string
+	GrpcClient    EdgeHubClient
+	sync.Mutex
+}
+
+type EdgeHubClient interface {
+	Collect(ctx context.Context, in *edgeHubPB.MetricFamilies, opts ...grpc.CallOption) (*edgeHubPB.Void, error)
+}
+
+// NewGRPCPushExporterServicer returns an exporter pushing metrics to
+// prometheus-edge-hubs at the given addresses.
+func NewGRPCPushExporterServicer(pushAddrs []string) protos.MetricsExporterServer {
+	conn, err := grpc.Dial(pushAddrs[0], getDialOptions()...)
+	if err != nil {
+		glog.Fatalf("Error creating grpc metrics exporter: %v", err)
+	}
+	conn.ResetConnectBackoff()
+	client := edgeHubPB.NewMetricsControllerClient(conn)
+
+	srv := &GRPCPushExporterServicer{
+		GrpcClient:    client,
+		PushAddresses: pushAddrs,
+	}
+	return srv
+}
+
+func (s *GRPCPushExporterServicer) Submit(ctx context.Context, req *protos.SubmitMetricsRequest) (*protos.SubmitMetricsResponse, error) {
+	s.Lock()
+	defer s.Unlock()
```

Comment:
what exactly is the shared resource you're protecting here? GRPC clients are threadsafe and it doesn't look like the push addresses are mutable

Reply:
oh good point I think I copypastad this from the old exporter